### ISSUE TITLE
Rspamd generates too much spam to be used with logcheck (log files monitoring system)

### ISSUE THREAD

**Issue by xeyownt** [rspamd/rspamd#1496]
I used rspamd on my Debian system, on which I also use logcheck to monitor the log files.

Currently I configured logcheck in paranoid mode, and I add the necessary rules to silent all harmless messages. Basically I configure logcheck such that I never receive any mails, except a few for rare events, or when there is really an issue (a backup that failed, a high load of unauthorised access...).

I did that for some time with rspamd, but now I'm just going to give up because rspamd generates much too much spam in the log files. At every update of rspamd, I receive lots of emails, with new kind of messages, etc. To circumvent this, I'll just setup a catch-all rule to tell logcheck to ignore all files from rspamd. But this defeats the purpose of using logcheck at the first place.

Is there / Is it possible to have a classification of rspamd messages, such as debug/info/warning/error/security/whatever? This way, I could configure logcheck to ignore unimportant message.

Also, do you have any recommendation for the filtering rules?

**Comment by fatalbanana** [comment]
The [log level](https://rspamd.com/doc/configuration/logging.html) of rspamd is configurable, so you could avoid outputting logs below `warning` or `error`.

Idea might be to add level as a variable in `log_format` so you could `[error]` tags.

**Comment by xeyownt** [comment]
Thanks for the suggestion.

I didn't try `warning` or `error` because I'd prefer to keep the current log level as is. I administer the server on a casual basis, and rspamd is certainly one of the most complex but also important daemon on the server. To be clear, it works perfectly, and worked almost out-of-the-box. But in case of troubles, I'd like to have access to enough information to analyse the source of the problem.

Adding a `$level` placeholder to variable `log_format` would be indeed interesting. Do I understand correctly that this is not available yet? I checked the rspamd doc page you refer to, but I must admit I don't understand it very well. For instance, I have

```
log_format =<<EOD
id: <$mid>,$if_qid{ qid: <$>,}$if_ip{ ip: $,}$if_user{ user: $,}$if_smtp_from{ from: <$>,}
(default: $is_spam ($action): [$scores] [$symbols_scores_params]),
len: $len, time: $time_real real, $time_virtual virtual, dns req: $dns_req,
digest: <$digest>$if_smtp_rcpts{, rcpts: <$>}$if_mime_rcpt{, mime_rcpt: <$>}
EOD
```

But in my logs I have many rspamd messages that do not match these format (like `task; accept_socket:`, `task; rspamd_message_parse:`...). It seems that `log_format` only corresponds to `task; rspamd_task_write_log: ` lines, but these are quite easy to filter out in logcheck.

So if there were a way to add the logging level, it should appear for every message sent by rspamd to the console.

**Comment by xeyownt** [comment]
bump

**Comment by xeyownt** [comment]
snif

**Comment by vstakhov** [comment]
There is log level `silent` for ages. Closing issue.

**Comment by xeyownt** [comment]
For info, log level `silent` does NOT solve this issue. Also, I would consider strange that the recommended setting for log level on a production system is basically **no logging**?!? Why then provide an option? As said, I'd like to have at least a setting with minimal logging that works with a tool like `logcheck`.

**Comment by moisseev** [comment]
@xeyownt It's not an Rspamd issue. Do you expect every program developer to write only log messages that match existing logcheck rules?
The correct approach is defining your own ignore rules in logcheck. Just put anything you want to suppress in your local logcheck rules file.

**Comment by xeyownt** [comment]
@moisseev Yes, I understand, and thanks for the suggestion. 

But in fact that's exactly what I am (was) trying to do. The problem, and the point of this ticket, is that the *format* and *layout* of the rspamd logging message keeps changing, so it's hard to come with a set of consistent rules. I did this for months, tracking any changes, trying to figure out some general pattern I could extract to separate important from not-so important messages, but finally I just had to give up.

IMO it would interesting (following fatalbanana suggestion, if I understood it well), to have within *each* message some *loglevel* we could parse to separate informative message from more important message in a consistent way.

Note that I don't want to disable logging because this is a server that I manage personally for a bigger team but I have basically zero experience in the domain. If a problem occur, I'd like to have the possibility to look at the logs to troubleshoot issues, if any.
